components/clp-py-utils/clp_py_utils/clp_metadata_db_utils.py

```diff
@@ -0,0 +1,125 @@
+        CREATE TABLE IF NOT EXISTS `{table_prefix}{ARCHIVE_TAGS_TABLE_SUFFIX}` (
+            `archive_id` VARCHAR(64) NOT NULL,
+            `tag_id` INT unsigned NOT NULL,
+            PRIMARY KEY (`archive_id`,`tag_id`),
+            FOREIGN KEY (`archive_id`) REFERENCES `{table_prefix}{ARCHIVES_TABLE_SUFFIX}` (`id`),
+            FOREIGN KEY (`tag_id`) REFERENCES `{table_prefix}{TAGS_TABLE_SUFFIX}` (`tag_id`)
+        )
+        """
+    )
+
+
+def _create_files_table(db_cursor, table_prefix: str) -> None:
+    db_cursor.execute(
+        f"""
+        CREATE TABLE IF NOT EXISTS `{table_prefix}{FILES_TABLE_SUFFIX}` (
+            `id` VARCHAR(64) NOT NULL,
+            `orig_file_id` VARCHAR(64) NOT NULL,
+            `path` VARCHAR(12288) NOT NULL,
+            `begin_timestamp` BIGINT NOT NULL,
+            `end_timestamp` BIGINT NOT NULL,
+            `num_uncompressed_bytes` BIGINT NOT NULL,
+            `begin_message_ix` BIGINT NOT NULL,
+            `num_messages` BIGINT NOT NULL,
+            `archive_id` VARCHAR(64) NOT NULL,
+            KEY `files_path` (path(768)) USING BTREE,
+            KEY `files_archive_id` (`archive_id`) USING BTREE,
+            PRIMARY KEY (`id`)
+        ) ROW_FORMAT=DYNAMIC
+        """
+    )
+
+
+def _create_column_metadata_table(db_cursor, table_prefix: str) -> None:
+    db_cursor.execute(
+        f"""
+        CREATE TABLE IF NOT EXISTS `{table_prefix}{COLUMN_METADATA_TABLE_SUFFIX}` (
+            `name` VARCHAR(512) NOT NULL,
+            `type` TINYINT NOT NULL,
+            PRIMARY KEY (`name`, `type`)
+        )
+        """
+    )
+
+
+def create_datasets_table(db_cursor, table_prefix: str) -> None:
+    """
+    Creates the table to store the datasets information (for clp-s only).
+    :param db_cursor: The database cursor to execute the table creation.
+    :param table_prefix: A string prefix applied to the dataset table name.
+    """
+    db_cursor.execute(
+        f"""
+        CREATE TABLE IF NOT EXISTS `{table_prefix}{DATASETS_TABLE_SUFFIX}` (
+            `name` VARCHAR(255) NOT NULL,
+            `archive_storage_directory` VARCHAR(4096) NOT NULL,
+            PRIMARY KEY (`name`)
+        )
+        """
+    )
+
+
+def create_metadata_db_tables(db_cursor, table_prefix: str, dataset: str | None = None) -> None:
+    """
+    Creates the standard set of SQL tables in the metadata database for recording compression tasks'
+    results.
+    :param db_cursor: The database cursor to execute the table creations.
+    :param table_prefix: A string prefix applied to all table names.
+    :param dataset: If provided, dataset-specific set of tables are created (for clp-s only).
+    """
+    if dataset is not None:
+        table_prefix = f"{table_prefix}{dataset}_"
+        _create_column_metadata_table(db_cursor, table_prefix)
+
+    _create_archives_table(db_cursor, table_prefix)
+    _create_tags_table(db_cursor, table_prefix)
+    _create_archive_tags_table(db_cursor, table_prefix)
```

Comment:
Sorry, this is my mistake---for these 3 tables, the way you had it before where you set the names outside and then passed them in was more sensible.